src/main/java/com/redhat/devtools/lsp4ij/client/features/LSPOnTypeFormattingFeature.java

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.client.features;
+
+import com.intellij.lang.LanguageFormatting;
+import com.intellij.psi.PsiFile;
+import com.redhat.devtools.lsp4ij.server.capabilities.OnTypeFormattingCapabilityRegistry;
+import org.eclipse.lsp4j.ServerCapabilities;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * LSP onTypeFormatting feature.
+ */
+@ApiStatus.Experimental
+public class LSPOnTypeFormattingFeature extends AbstractLSPDocumentFeature {
+
+    private OnTypeFormattingCapabilityRegistry onTypeFormattingCapabilityRegistry;
+
+    @Override
+    public boolean isEnabled(@NotNull PsiFile file) {
+        // Need to perform the same check as in LSPFormattingFeature.isEnabled() to ensure that LSP formatting should be used
+        if (!getClientFeatures().getFormattingFeature().isExistingFormatterOverrideable(file) &&
```

Comment:
I believe this is the check that you'd mentioned needed to occur, correct? Let me know if it's not correct? Obviously I could just call `getClientFeatures().getFormattingFeature().isEnabled(file)` instead, but if that method were overridden in a derived class, we might disable on-type formatting when only the full formatter should be disabled. Let me know if you'd prefer that I just simplify it to that, though.

Reply:
Great! Thanks.